kratos/tests/cpp_tests/strategies/schemes/test_residual_based_adjoint_bossak_scheme.cpp

```diff
@@ -149,17 +149,42 @@ namespace NonLinearSpringMassDamper
  * sensitivity analysis for nonlinear transient problems.
  *
  *  |                _____                 _____
- *  |---[ Damper ]--|mass1|---[ Damper ]--|mass2|
+ *  |---[ Damper ]--|mass1|---[ Damper ]--|mass2|--> F
  *  |-----/\/\/\----|_____|-----/\/\/\----|_____|
  *  |
  *
  * Spring force: fe = x + stiffness * x^3
  * Damper force: fc = damping * x'
  *
- * Momentum equations:
+ * Following point load is only used in the Condition test case, otherwise this is zero
+ * F = 1.0 * x^3 - 2.0 * damping * x' - mass1 * x''
+ *
+ * Momentum equations: (For element test case)
  * mass1 * acc1 + 2 * damping * vel1 - damping * vel2 + 2 * disp1 - disp2 + stiffness * disp1^3 - stiffness * (disp2 - disp1)^3 = 0,
  * mass2 * acc2 - damping * vel1 + damping * vel2 - disp1 + disp2 + stiffness * (disp2 - disp1)^3 = 0.
+ *
+ * Momentum equations: (For conditions test case)
+ * mass1 * acc1 + 2 * damping * vel1 - damping * vel2 + 2 * disp1 - disp2 + stiffness * disp1^3 - stiffness * (disp2 - disp1)^3 = 0,
+ * mass2 * acc2 - damping * vel1 + damping * vel2 - disp1 + disp2 + stiffness * (disp2 - disp1)^3 - 1.0 * disp2^3 + 2.0 * damping * vel2 + mass1 * acc2 = 0.
+ *
+ * The above mentioned test is calculating sensitivity on a quantity inside an element (i.e. sensitivity w.r.t. stiffness).
+ * ResidualBasedAdjointBossakScheme is capable of formulating adjoint system of equations even if Elements and Conditions are
+ * both using this same quantity inside. But since this test only pass SCALAR_SENSITIVITY variable to sensitivity builder under
+ * elements container, when it builds up sensitivities it does not get the contributions coming from the conditions.
+ * Therefore, this test does not use `stiffness` inside the conditions to avoid it to have any contributions for partial
+ * sensitivity matrix.
+ *
+ * In the case, if nodal sensitivities are calculated, then sensitivity builder calls to get contributions for partial sensitivity
+ * from Elements as well as Conditions. Therefore, Elements and Condtions can be safely used without any trouble in the case of
+ * nodal sensitivities.
```

Comment:
Maybe you move the comments towards ``CalculateSensitivityMatrix`` of the condition

Reply:
done :)